home/tests/integration/views/web/test_data.py

```diff
@@ -0,0 +1,174 @@
+        }
+        contest_id = next(ContestGenerator().generate(1, **params)).pk
+
+        response = c.get(
+            "/data/users_agg.csv",
+            {
+                "contest_id": contest_id,
+            },
+        )
+
+        self.assertEqual(200, response.status_code)
+        self.assertEqual("text/csv", response["Content-Type"])
+        content = response.content.decode("utf-8")
+        reader = csv.DictReader(io.StringIO(content))
+        rows = list(reader)
+        headers = reader.fieldnames
+        self.assertIn("email", headers)
+        self.assertIn("total_steps", headers)
+        self.assertEqual(2, len(rows))
+        for row in rows:
+            # Daily walks: [7, 8, 9, 10]
+            self.assertEqual(int(row["num_daily_walks"]), 4)
+            # Intentional walks: [8, 10]
+            self.assertEqual(int(row["num_recorded_walks"]), 2)
+
+    # Test csv response of daily walks
+    def test_daily_walks_csv_view(self):
+        c = Client()
+        self.assertTrue(self.login(c))
+        start_date = date(3000, 3, 7)
+        end_date = date(3000, 3, 14)
+        response = c.get(
+            "/data/daily_walks.csv",
+            {
+                "start_date": start_date.isoformat(),
+                "end_date": end_date.isoformat(),
+            },
+        )
+
+        self.assertEqual(200, response.status_code)
+        self.assertEqual("text/csv", response["Content-Type"])
+        content = response.content.decode("utf-8")
+        reader = csv.DictReader(io.StringIO(content))
+        rows = list(reader)
+        headers = reader.fieldnames
+        self.assertIn("date", headers)
+        self.assertIn("steps", headers)
+
+        grouped_rows = self.group_by(rows, "email")
+        for user, walks in grouped_rows.items():
+            self.assertEqual(4, len(walks))  # [7, 8, 9, 10]
+
+            for walk in walks:
+                self.assertGreaterEqual(date.fromisoformat(walk["date"]), start_date)
+                self.assertLessEqual(date.fromisoformat(walk["date"]), end_date)
+
+    # Test csv response of intentional (recorded) walks
+    def test_intentional_walks_csv_view(self):
+        c = Client()
+        self.assertTrue(self.login(c))
+        start_date = date(3000, 3, 7)
+        end_date = date(3000, 3, 14)
+        response = c.get(
+            "/data/intentional_walks.csv",
+            {
+                "start_date": start_date.isoformat(),
+                "end_date": end_date.isoformat(),
+            },
+        )
+
+        self.assertEqual("text/csv", response["Content-Type"])
+        self.assertEqual(200, response.status_code)
+        content = response.content.decode("utf-8")
+        reader = csv.DictReader(io.StringIO(content))
+        rows = list(reader)
+        headers = reader.fieldnames
+        self.assertIn("event_id", headers)
+        self.assertIn("steps", headers)
+
+        grouped_rows = self.group_by(rows, "email")
```

Comment:
This groups the intentional walks by user but does not combine the rows. It essentially creates a map like
```
{
  "user1@email.us": [
    [row containing user1 walk on the 8th],
    [row containing user1 walk on the 10th],
  "user2@email.us": [
    [row containing user2 walk on the 8th],
    [row containing user2 walk on the 10th],
  ]
}
```